examples/models/object_detection/retina_net/pascal_voc/train.py

```diff
@@ -0,0 +1,567 @@
+    backbone_weights="imagenet",
+    # include_rescaling tells the model whether your input images are in the default
+    # pixel range (0, 255) or if you have already rescaled your inputs to the range
+    # (0, 1).  In our case, we feed our model images with inputs in the range (0, 255).
+    include_rescaling=True,
+    # Typically, you'll want to set this to False when training a real model.
+    # evaluate_train_time_metrics=True makes `train_step()` incompatible with TPU,
+    # and also causes a massive performance hit.  It can, however be useful to produce
+    # train time metrics when debugging your model training pipeline.
+    evaluate_train_time_metrics=False,
+)
+# Fine-tuning a RetinaNet is as simple as setting backbone.trainable = False
+model.backbone.trainable = False
+
+metrics = [
+    keras_cv.metrics.COCOMeanAveragePrecision(
+        class_ids=range(21),
+        bounding_box_format="xywh",
+        name="Mean Average Precision",
+    ),
+    keras_cv.metrics.COCORecall(
+        class_ids=range(21),
+        bounding_box_format="xywh",
+        max_detections=100,
+        name="Recall",
+    ),
+]
+optimizer = tf.optimizers.SGD(global_clipnorm=10.0)
+model.compile(
+    classification_loss="focal",
+    box_loss="smoothl1",
+    optimizer=optimizer,
+    metrics=metrics,
+)
+
+callbacks = [
+    keras.callbacks.TensorBoard(log_dir="logs"),
+    keras.callbacks.ReduceLROnPlateau(patience=5),
+    keras.callbacks.ModelCheckpoint(CHECKPOINT_PATH, save_weights_only=True),
+]
+
+history = model.fit(
+    train_ds,
+    validation_data=eval_ds.take(10),
+    epochs=100,
+    callbacks=callbacks,
+)
+
+"""
+## Training Process Visualization
+"""
+metrics = history.history
+metrics_to_plot = {
+    "Train": metrics["loss"],
+    "Validation": metrics["val_loss"],
+    "Train Box Loss": metrics["box_loss"],
+    "Validation Box Loss": metrics["val_box_loss"],
+    "Train Classification Loss": metrics["classification_loss"],
+    "Validation Classification Loss": metrics["val_classification_loss"],
+}
+
+visualization.line_plot(
+    data=metrics_to_plot,
+    title="Loss",
+    xlabel="Epochs",
+    ylabel="Loss",
+    transparent=True,
+    path="learning_curve.png",
+)
+
+metrics_to_plot = {
+    "Mean Average Precision": metrics["val_MaP"],
+    "Recall": metrics["val_Recall"],
+}
+visualization.line_plot(
+    data=metrics_to_plot,
+    title="Loss",
+    xlabel="Epochs",
+    ylabel="Loss",
+    transparent=True,
+    path="coco_metrics.png",
+)
+
+"""
+## Inference
+"""
+
+images, y_true = next(iter(eval_ds.take(1)))
+y_pred = model.predict(images)
+visualization.plot_bounding_box_gallery(
+    images,
+    value_range=(0, 255),
+    bounding_box_format="xywh",
+    y_true=y_true,
+    y_pred=y_pred,
+    scale=4,
+    rows=2,
+    cols=4,
+    show=True,
+    thickness=4,
+    font_scale=1,
+    class_mapping=class_mapping,
+    path="inference.png",
+)
+
+"""
+## Evaluation
+"""
+
+coco_suite = [
```

Comment:
why not just model.fit(train_ds, validation_data=eval_ds)?

Reply:
if you evaluate the coco metrics during training after each epoch it takes way too long to run.

Reply:
Maybe we should have some canonical training workflow like:
```
model.compile(..., metrics=[cocomAP_suite])
model.fit(train_ds, validation_data=eval_ds, epochs=?)
model.evaluate(eval_ds)
```
Though with cocomAP accuracy still under investigation, it's definitely ok not to include them for now. No strong opinion here.